src/core.js

```diff
@@ -617,15 +617,19 @@ JSONEditor.prototype = {
       }
       var ref = fetchUrl + refObj.$ref
       if (!this.refs[ref]) ref = fetchUrl + decodeURIComponent(refObj.$ref)
+      if (!this.refs[ref]) { // if reference not found
+        console.warn("reference:'" + ref + "' not found!")
+        break
+      }
       if (recurseAllOf) {
         if (this.refs[ref].hasOwnProperty('allOf')) {
-          var allOf = this.refs[ref].allOf
-          for (var i = 0; i < allOf.length; i++) {
-            allOf[i] = this.expandRefs(allOf[i], true)
-          }
+          this.refs[ref].allOf = this.refs[ref].allOf.map(function (one) {
+            return self.expandRefs(one, true)
+          })
```

Comment:
since the `$extend` method didn't deepcopy the array, the content of array should avoid to be revised

Reply:
Perhaps the best solution would be to replace the $extend with a better clone function. See #506 
Unfortunately @sirockin won't be around to implement it. So if you want to improve/replace the $extend with a more complete deepcopy function, please do,
This would probably solve several issues, specially array related.